src/cmd/src/cli/export.rs

```diff
@@ -88,59 +95,84 @@ impl ExportCommand {
                 addr: self.addr.clone(),
             })?;
         let (catalog, schema) = split_database(&self.database)?;
-        let mut database_client = Database::new(
-            catalog.clone(),
-            schema.clone().unwrap_or(DEFAULT_SCHEMA_NAME.to_string()),
-            client,
-        );
 
-        if let Some(auth_basic) = &self.auth_basic {
+        let auth_header = if let Some(auth_basic) = &self.auth_basic {
             let (username, password) = auth_basic.split_once(':').context(IllegalConfigSnafu {
                 msg: "auth_basic cannot be split by ':'".to_string(),
             })?;
-            database_client.set_auth(AuthScheme::Basic(Basic {
-                username: username.to_string(),
-                password: password.to_string(),
-            }));
-        }
+            Some(AuthHeader {
+                auth_scheme: Some(AuthScheme::Basic(Basic {
+                    username: username.to_string(),
+                    password: password.to_string(),
+                })),
+            })
+        } else {
+            None
+        };
 
         Ok(Instance::new(Box::new(Export {
-            client: database_client,
             catalog,
             schema,
             output_dir: self.output_dir.clone(),
             parallelism: self.export_jobs,
             target: self.target.clone(),
+            auth_header,
+            region_requester: RegionRequester::new(client),
         })))
     }
 }
 
 pub struct Export {
-    client: Database,
     catalog: String,
     schema: Option<String>,
     output_dir: String,
     parallelism: usize,
     target: ExportTarget,
+    auth_header: Option<AuthHeader>,
+    region_requester: RegionRequester,
 }
 
 impl Export {
+    /// Execute a sql query.
+    async fn sql(&self, sql: &str, catalog: String, schema: String) -> Result<Output> {
+        let request = Request::Query(QueryRequest {
+            query: Some(Query::Sql(sql.to_string())),
+        });
```

Comment:
We plan to remove `Request::Query` in #3799.

Perhaps we should try to find an alternative way to implement the same feature here.

cc @MichaelScofield

Reply:
I'm thinking of using http API instead.

Reply:
@MichaelScofield 

I tried `reqwest::Request` but return value `reqwest::Response` is not a stream. Is this direction correct?

If I use following code
https://github.com/GreptimeTeam/greptimedb/blob/e154dc5fd46938f24f0f91363a608f75b107a929/src/servers/src/http/handler.rs#L67-L75
how to get `frontend instance handler` to construct `ApiState`
```rust
pub struct ApiState {
    pub sql_handler: ServerSqlQueryHandlerRef,
    pub script_handler: Option<ScriptHandlerRef>,
}
```

Reply:
The `ApiState` is used in http server side. Stream is not necessary here. You can find some example codes in "tests-integration/tests/http.rs".